src/main/java/org/opensearch/knn/plugin/stats/KNNStats.java

```diff
@@ -75,4 +76,73 @@ private Map<String, KNNStat<?>> getClusterOrNodeStats(Boolean getClusterStats) {
         }
         return statsMap;
     }
+
+    private Map<String, KNNStat<?>> buildStatsMap() {
+        return ImmutableMap.<String, KNNStat<?>>builder()
```

Comment:
let's break this function into smaller chunks of private functions which tells what stats we are adding. So that it becomes easier to debug and read.

Reply:
Sure, will update.